powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/service/CredentialService.java

```diff
@@ -638,14 +657,16 @@ public CredentialEntity findCredential(CredentialDefinitionEntity credentialDefi
      * @param credentialValue Credential value, use null for generated credential value.
      * @param timestampExpires Expiration timestamp for case when expiration timestamp is overridden.
      * @param validationMode Credential validation mode.
+     * @param source Credential source, either local or managed by some external proxy.
+     * @param target Credential target, the credentials can upon change switch the source.
      * @throws InvalidConfigurationException Thrown when Next Step configuration is invalid.
      * @throws CredentialValidationFailedException Thrown when credential validation fails.
      * @throws InvalidRequestException Thrown when request is invalid.
      * @throws EncryptionException Thrown when encryption or decryption fails.
      */
     public CredentialSecretDetail createCredential(UserIdentityEntity user, CredentialDefinitionEntity credentialDefinition,
                                                    CredentialType credentialType, String username, String credentialValue,
-                                                   Date timestampExpires, CredentialValidationMode validationMode) throws InvalidConfigurationException, CredentialValidationFailedException, InvalidRequestException, EncryptionException {
+                                                   Date timestampExpires, CredentialValidationMode validationMode, CredentialLocation source, CredentialLocation target) throws InvalidConfigurationException, CredentialValidationFailedException, InvalidRequestException, EncryptionException {
```

Comment:
The long parameter list of this method is not a good practice, but unfortunately, I do not see any easy way to make it nicer 😞